File: python/49c_1618A_polycarpAndSumsOfSubsequences.py
def solve(b):
    for i in range(7):
        b[i] = int(b[i])
    b.sort()
    a = []
    a.append(b[0])
    a.append(b[1])
    a.append(b[-1] - b[0] - b[1])
    return a

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        arr = input().split()
        ans.append(solve(arr))
    for i in ans:
        for j in i:
            print(j, end = " ")
        print()

if __name__ == "__main__":
    main()

File: cpp/49c_1618A_polycarpAndSumsOfSubsequences.cpp
#include<bits/stdc++.h>

std::vector<int> solve(std::vector<int> b){
    std::vector<int> a;
    sort(b.begin(), b.end());
    a.push_back(b[0]);
    a.push_back(b[1]);
    a.push_back(b[6]-b[0]-b[1]);
    return a;
}

int main(){
    int t, temp;
    std::vector<std::vector<int>> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::vector<int> b;
        for(int j = 0; j < 7; j++){
            std::cin>>temp;
            b.push_back(temp);
        }
        ans.push_back(solve(b));
    }
    for (auto i : ans){
        for(auto j : i)
            std::cout<<j<<" ";
        std::cout<<"\n";
    }
    return 0;
}